### PR TITLE
Issue #15340: created InputFormatted file for section 7.1.2 Paragraphs

### PR DESCRIPTION
**Pull request by Zopsss**
#15340

### CODE REVIEW

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java
Comment (rdiachenko) [inline src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java:100]: Is it a bug in the formatter?

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java
Comment (rdiachenko) [inline src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java:113]: Is it a bug in the formatter?

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java:100]: Yes, I had the same question.
But formatter might not bother about this at all. 
@Zopsss , please confirm.

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java:113]: I think yes, rule [7.1.2 Paragraphs](https://google.github.io/styleguide/javaguide.html#s7.1.2-javadoc-paragraphs) says: 

> One blank line—that is, a line containing only the aligned leading asterisk (*)—appears between paragraphs, and before the group of block tags if present.

"group of block tags" can be @param @see @return, etc. So there should be a blank line between `<p>` and `@see`

good catch. Will open an issue regarding this on formatter's GH

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java:100]: [7.1.2 Paragraphs](https://google.github.io/styleguide/javaguide.html#s7.1.2-javadoc-paragraphs) says:

> Each paragraph except the first has <p> immediately before the first word, with no space after it.

so `<p>` should be placed immediately before `/WARN/ Some Javadoc.`, something like:

```
<p>/WARN/ Some Javadoc.
```

so this might be a formatter's bug, will open an issue regarding this on formatter's GH

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java:100]: formatter do care a bit about P tags - https://github.com/google/google-java-format/issues?q=is%3Aissue+p+tag+is%3Aclosed

let close issue on them and let move forward with merge of this PR.

**Review by romani — APPROVED** [review]
ok to merge after issue is created against formatter

**Review by rdiachenko — APPROVED** [review]
lgtm

@Zopsss please reply with links to the created issues against the formatter, so we could merge this PR.

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java:100]: it is grey area, there is no words, so there could be multiple of interpretations of

> Each paragraph except the first has `<p>` immediately before the first word, with no space after it.


https://github.com/google/google-java-format/issues/1150

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java:113]: this is not valid javadoc.

```
rivanov@p5510:/var/tmp$ cat Test.java
public class Test {
    /**
    * The client's first name.
    *
    * @see <a
    *     href="http://www.gwtproject.org/doc/latest/DevGuideOrganizingProjects.html#DevGuideModules">
    *     Documentation about
    *     <p>GWT emulated source</a>
    */
  public String thirdName;
}
rivanov@p5510:/var/tmp$ 
rivanov@p5510:/var/tmp$ 
rivanov@p5510:/var/tmp$ javadoc Test.java
Loading source file Test.java...
Constructing Javadoc information...
Standard Doclet version 11.0.14.1
Building tree for all the packages and classes...
Generating ./Test.html...
Test.java:8: error: block element not allowed within inline element <a>: p
    *     <p>GWT emulated source</a>
          ^
Generating ./package-summary.html...
Generating ./package-tree.html...
Generating ./constant-values.html...
Building index for all the packages and classes...
Generating ./overview-tree.html...
Generating ./index-all.html...
Building index for all classes...
Generating ./allclasses-index.html...
Generating ./allpackages-index.html...
Generating ./deprecated-list.html...
Building index for all classes...
Generating ./allclasses.html...
Generating ./allclasses.html...
Generating ./index.html...
Generating ./help-doc.html...
1 error

```


so it is not a defect of formater.

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java:113]: it is not a defect in Checkstyle as as have no validation for this at all for now. It should be some new Check, but let not create it now, let wait for users feedback.

**Pull request merged**
